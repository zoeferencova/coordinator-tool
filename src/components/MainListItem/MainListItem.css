.table-row i {
    font-size: 17px;
    margin: 0 7px;
    color: gray;
}

div .table-body-cell {
    padding: 10px 0;
}

.proj-cell, .adv-cell, .notes-cell {
    max-width: 136px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

div .notes-cell, div .adv-cell, div .proj-cell {
    padding: 0 10px;
}

.hide-list {
    display: none;
}

.show-list {
    display: block;
}

.status-icon i {
    font-size: 20px;
}

.icon-list li {
    list-style: none;
    border-bottom: 1px solid gray;
    padding: 6px;
    font-size: 12px;
}

.icon-list {
    position: absolute;
    background: lightgray;
    padding: 6px;
    text-align: left;
    margin: 3px 0px 0px 25px;
    border-radius: 5px;
}

.icon-list .last {
    border-bottom: none;
}

.status-icon:hover, .icon-list:hover {
    cursor: pointer;
}

@media only screen and (max-width: 1020px) {
    .table-row i {
        margin: 0 1px;
    }
}

.popup-overlay {
    background-color: black;
    opacity: 50%;
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 20;
}

.popup {
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 50;
}

.popup-container  {
    top: 50vh;
    margin: 0 auto;
    position: relative;
    width: 85%;
    opacity: 100%;
    z-index: 30;
    padding: none;
}

.popup-container > div {
    background-color: white;
    border: none;
    border-radius: 10px;
}

.popup-container .last {
    border-bottom: none;
}

.popup-container > div > div, .popup-link {
    border-bottom: 0.5px solid lightgray;
    padding: 10px 0;
}

.popup-container a {
    text-decoration: none;
    color: black;
}

.popup-container .cancel {
    margin-top: 15px;
    padding: 17px 0;
}




