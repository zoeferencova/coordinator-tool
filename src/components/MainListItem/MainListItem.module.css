.hideList {
    display: none;
}

.showList {
    display: block;
}

.icon i {
    color: #9299a8;
    cursor: pointer;
}

.statusSelect {
    width: 62px;
    font-size: 10px;
    border-radius: 5px;
    border: none;
    background-color: #e1e7f7;
    padding: 2px;
    cursor: pointer;
}

.pending {
    background-color: #e1f2e6;
    color: #2c7d43;
}

.popupOverlay {
    background-color: black;
    opacity: 0.5 !important;
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 20;
}

.popup {
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    left: 0;
    z-index: 50;
}

.popupContainer {
    top: 50vh;
    margin: 0 auto;
    position: relative;
    width: 85%;
    opacity: 1 !important;
    z-index: 30;
    padding: none;
}

.popupContainer > div {
    background-color: white;
    border: none;
    border-radius: 10px;
}

.popupContainer .last {
    border-bottom: none;
}

.popupContainer > div > div, .popupLink {
    border-bottom: 0.5px solid lightgray;
    padding: 10px 0;
}

.popupContainer a {
    text-decoration: none;
    color: black;
}

.popupContainer .cancel {
    margin-top: 15px;
    padding: 17px 0;
}



