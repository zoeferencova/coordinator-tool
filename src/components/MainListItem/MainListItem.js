import React from 'react';
import AppContext from '../../contexts/contexts'
import './MainListItem.css'
import { Link } from 'react-router-dom';
import ReactDOM from 'react-dom'
import config from '../../config';

const icons = {
    'none': <i className="far fa-circle"></i>,
    'reached': <i className="fas fa-arrow-circle-left"></i>,
    'completed': <i className="fas fa-check-circle"></i>
}

export default class MainListItem extends React.Component {
    static contextType = AppContext;
    
    state = {
        expanded: false
    }

    handleStatusClick(status, id) {
        fetch(`${config.API_ENDPOINT}/list/${id}`, {
            method: 'PATCH',
            headers: {
                'content-type': 'application/json',
                'Authorization': `Bearer ${window.sessionStorage.getItem(config.TOKEN_KEY)}`
            },
            body: JSON.stringify({ status })
        })
            .then(res => this.context.updateItemStatus(id, status))

        if (this.state.expanded === true) {
            this.setState({ expanded: false })
        } else {
            this.setState({ expanded: true })
        }
    }

    handleDeleteItem(e) {
        e.preventDefault();
        const itemId = ReactDOM.findDOMNode(e.target).parentNode.getAttribute('itemkey')
        fetch(`${config.API_ENDPOINT}/list/${itemId}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'Authorization': `Bearer ${window.sessionStorage.getItem(config.TOKEN_KEY)}`
            }
        })
        this.context.deleteItem(itemId)
    }

    render() {
        const { id, project, project_url, advisor, advisor_url, pm_name, pm_email, notes, status, date_created } = this.props;
        return (
            <div className="table-row row">
                <div className="table-body-cell hide-mobile"><input type="checkbox" id="list-checkbox" onChange={e => this.props.setChecked(id)}></input></div>
                <div className="table-body-cell">
                    <span className="status-icon" onClick={() => this.state.expanded === true ? this.setState({ expanded: false }) : this.setState({ expanded: true })}>{icons[status]}</span>
                    <ul className={`icon-list hide-list ${this.state.expanded && 'show-list'}`}>
                        <li onClick={() => this.handleStatusClick('none', id)}>{icons['none']} None</li>
                        <li onClick={() => this.handleStatusClick('reached', id)}>{icons['reached']} Reached Out</li>
                        <li className='last' onClick={() => this.handleStatusClick('completed', id)}>{icons['completed']} Completed</li>
                    </ul>
                </div>
                <div className="table-body-cell proj-cell">{project_url !== '' ? <a href={project_url} target="_blank">{project}</a> : project}</div>
                <div className="table-body-cell proj-cell">{advisor_url !== '' ? <a href={advisor_url} target="_blank">{advisor}</a> : advisor}</div>
                <div className="table-body-cell hide-mobile">{pm_name}</div>
                <div className="table-body-cell hide-mobile">{date_created}</div>
                <div className="table-body-cell notes-cell hide-mobile">{notes}</div>
                <div className="table-body-cell hide-mobile">
                    <button onClick={() => this.props.openEmailForm(project, advisor, pm_name, pm_email)}><i className="fas fa-envelope"></i></button>
                    <Link to={{pathname:`/edit-item/${id}`, itemProps: {project, advisor, pm_name, notes}}} ><button onClick={this.handleEditItem}><i className="fas fa-edit"></i></button></Link>
                    <button itemkey={id} onClick={e => this.handleDeleteItem(e)}><i className="fas fa-trash-alt"></i></button>
                </div>
                <div className="table-body-cell hide-desktop">
                    <div 
                        onClick={() => this.state.popup 
                            ? this.setState({ popup: false }) 
                            : this.setState({popup: true})}
                    >
                        <i className="fas fa-ellipsis-h"></i>
                    </div>
                    <div className={`popup ${this.state.popup ? 'show' : 'hidden'}`}>
                        <div className="popup-overlay" onClick={() => this.setState({ popup: false })}></div>
                        <div className='popup-container'>
                            <div itemkey={id}>
                                <div>PM: {pm_name}</div>
                                <div>Date: {date_created}</div>
                                {this.props.notes && <div>Notes: {notes}</div>}
                                <div onClick={() => this.props.openEmailForm(project, advisor, pm_name, pm_email)} >Email...</div>
                                <Link to={{pathname:`/edit-item/${id}`, itemProps: {project, advisor, pm_name, notes}}} ><div className="popup-link" onClick={this.handleEditItem}>Edit...</div></Link>
                                <div className='last' itemkey={id} onClick={e => this.handleDeleteItem(e)}>Delete</div>
                            </div>
    
                            <div className='cancel' onClick={() => this.setState({ popup: false })}>Cancel</div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}