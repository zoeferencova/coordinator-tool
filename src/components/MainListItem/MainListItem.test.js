import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainListItem from './MainListItem'

describe(`MainListItem component`, () => {
  it('renders a .MainListItem by default', () => {
    const wrapper = shallow(<MainListItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})