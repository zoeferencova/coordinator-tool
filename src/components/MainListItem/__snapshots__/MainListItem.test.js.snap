// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainListItem component renders a .MainListItem by default 1`] = `
<div
  className="tableRow"
>
  <div
    className="tableBodyCell hideMobile check"
  >
    <input
      id="list-checkbox"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
  <div
    className="tableBodyCell project"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    />
  </div>
  <div
    className="tableBodyCell advisor"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    />
  </div>
  <div
    className="tableBodyCell hideMobile pm"
  />
  <div
    className="tableBodyCell hideMobile date"
  />
  <div
    className="tableBodyCell hideMobile notes"
  />
  <div
    className="tableBodyCell status"
  >
    <select
      className="statusSelect "
      onChange={[Function]}
    >
      <option
        value="none"
      />
      <option
        value="reached"
      >
        Pending
      </option>
      <option
        value="completed"
      >
        Completed
      </option>
    </select>
  </div>
  <div
    className="tableBodyCell hideMobile actions"
  >
    <button
      className="icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-envelope"
      />
    </button>
    <Link
      to={
        Object {
          "itemProps": Object {
            "advisor": undefined,
            "notes": undefined,
            "pm_name": undefined,
            "project": undefined,
          },
          "pathname": "/edit-item/undefined",
        }
      }
    >
      <button
        className="icon"
      >
        <i
          className="fas fa-edit"
        />
      </button>
    </Link>
    <button
      className="icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-trash"
      />
    </button>
  </div>
  <div
    className="tableBodyCell hideDesktop mobileActions"
  >
    <div
      onClick={[Function]}
    >
      <i
        className="fas fa-ellipsis-h pointer"
      />
    </div>
    <div
      className="popup hidden"
    >
      <div
        className="popupOverlay"
        onClick={[Function]}
      />
      <div
        className="popupContainer"
      >
        <div>
          <div>
            PM: 
          </div>
          <div>
            Date: 
          </div>
          <div
            className="pointer"
            onClick={[Function]}
          >
            Email...
          </div>
          <Link
            to={
              Object {
                "itemProps": Object {
                  "advisor": undefined,
                  "notes": undefined,
                  "pm_name": undefined,
                  "project": undefined,
                },
                "pathname": "/edit-item/undefined",
              }
            }
          >
            <div
              className="popupLink"
            >
              Edit...
            </div>
          </Link>
          <div
            className="last pointer"
            onClick={[Function]}
          >
            Delete
          </div>
        </div>
        <div
          className="cancel pointer"
          onClick={[Function]}
        >
          Cancel
        </div>
      </div>
    </div>
  </div>
</div>
`;
